### ISSUE TITLE
`no-array-reduce` - doesn't report optional chaining

### ISSUE THREAD

**Issue by ipanasenko** [sindresorhus/eslint-plugin-unicorn#2720]
Title is pretty self-explanatory. `no-array-reduce` doesn't report optional chaining calls of `reduce`, e.g. `array?.reduce()`

Rule: `no-array-reduce`

```js
array.reduce((str, item) => str += item, "") // reports correctly as error
array?.reduce((str, item) => str += item, "") // doesn't report this one
```

I see in rule's code `optionalMember: false` is explicitly passed. So, is this really a bug or an intended behaviour?
I've created a PR with new test cases - they fail: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2721

Thanks!

**Comment by ipanasenko** [comment]
All mutating methods rules also don't check optional chaining, did I get that right? E.g. `no-array-sort`

**Comment by ipanasenko** [comment]
Enabled reporting of optionally chained and called `.reduce()` in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2721
